cdrs-tokio-helpers-derive/src/common.rs

```diff
@@ -204,6 +216,21 @@ fn as_rust(ty: &Type, val: TokenStream) -> TokenStream {
     match get_ident_string(&cdrs_type).as_str() {
         "Blob" | "String" | "bool" | "i64" | "i32" | "i16" | "i8" | "f64" | "f32" | "IpAddr"
         | "Uuid" | "Timespec" | "Decimal" | "PrimitiveDateTime" => val,
+        "Set" => {
+            let set_type = get_ident_params_string(ty);
+            let inter_rust_type = get_cdrs_type(&set_type);
+            let decoded_item = as_rust(&set_type, quote! {item});
+            quote! {
+              {
+                let inner: std::collections::BTreeSet<#inter_rust_type> = #val.as_r_type()?;
```

Comment:
I don't think a BTreeSet is a good default - it's a rather specific collection, where a HashSet is more widely used.

Reply:
`HashSet` can't be hashed so it needed to be `BTreeSet` so it was possible to represent nested sets.

Reply:
Ok, that's a good point.

Reply:
Also, `BTreeSet` can be generic over an allocator, so we need to make sure it works too.